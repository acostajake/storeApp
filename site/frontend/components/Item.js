import Link from 'next/link';
import PropTypes from 'prop-types';
import React from 'react';

import formatMoney from '../lib/formatMoney';
import AddToCart from './AddToCart';
import ItemStyles from './styles/ItemStyles';
import DeleteItem from './DeleteItem';
import PriceTag from './styles/PriceTag';
import Title from './styles/Title';

const Item = (props) => {
	const {
		item: { desc, id, image, price, title },
	} = props;
	return (
		<ItemStyles>
			{image && <img src={image} alt={title} />}
			<Title>
				<Link
					href={{
						pathname: '/item',
						query: { id },
					}}
				>
					<a>{title}</a>
				</Link>
			</Title>
			<PriceTag>{formatMoney(price)}</PriceTag>
			<p>{desc}</p>
			<div className='buttonList'>
				<Link href={{ pathname: 'update', query: { id } }}>
					<a>Edit Item</a>
				</Link>
				<AddToCart id={id} />
				<DeleteItem id={id}>Delete Item</DeleteItem>
			</div>
		</ItemStyles>
	);
};

Item.propTypes = {
	item: PropTypes.shape({
		desc: PropTypes.string,
		id: PropTypes.string,
		image: PropTypes.string,
		price: PropTypes.number,
		title: PropTypes.string,
	}).isRequired,
};

export default Item;
