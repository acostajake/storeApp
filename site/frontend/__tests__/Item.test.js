import { shallow } from 'enzyme';
import formatMoney from '../lib/formatMoney';
import Item from '../components/Item';
import toJSON from 'enzyme-to-json';

const demoItem = {
	id: 'demo123',
	name: 'Sample Item',
	price: 3000,
	desc: 'First item',
	image: 'demo.png',
	largeImg: 'largeDemo.png',
};

const demoWrapper = shallow(<Item item={demoItem} />);

describe('<Item/>', () => {
	it('handles the price', () => {
		const priceTag = demoWrapper.find('PriceTag');
		expect(priceTag).toHaveLength(1);
		expect(priceTag.children().text()).toEqual(formatMoney(demoItem.price));
	});
	it('displays the item image', () => {
		const img = demoWrapper.find('img');
		expect(img).toHaveLength(1);
		expect(img.props().src).toEqual(demoItem.image);
		expect(img.props().alt).toEqual(demoItem.title);
	});
	it('displays buttons', () => {
		const buttons = demoWrapper.find('.buttonList');
		expect(buttons.children()).toHaveLength(3);
		expect(buttons.find('Link').exists()).toBe(true);
		expect(buttons.find('AddToCart').exists()).toBe(true);
		expect(buttons.find('DeleteItem').exists()).toBe(true);
	});
});
