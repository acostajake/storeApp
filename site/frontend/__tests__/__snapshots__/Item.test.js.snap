// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    src="demo.png"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "demo123",
          },
        }
      }
    >
      <a />
    </Link>
  </Title>
  <PriceTag>
    $30
  </PriceTag>
  <p>
    First item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "demo123",
          },
        }
      }
    >
      <a>
        Edit Item
      </a>
    </Link>
    <AddToCart
      id="demo123"
    />
    <DeleteItem
      id="demo123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
